orientdb/src/main/scala/akka/stream/alpakka/orientdb/OrientDBSourceStage.scala

```diff
@@ -0,0 +1,158 @@
+    with OutHandler {
+
+  private val responseHandler = getAsyncCallback[List[T]](handleResponse)
+  private val failureHandler = getAsyncCallback[Throwable](handleFailure)
+
+  def sendOSQLRequest(): Unit = {
+    val client = settings.oDatabasePool.acquire()
+
+    if (clazz.isEmpty) {
+      try {
+        if (query != null) {
+          client.query[java.util.List[T]](new OSQLNonBlockingQuery[T](query, new OSQLCallback))
+        } else {
+          client.query[java.util.List[T]](
+            new OSQLNonBlockingQuery[T](
+              s"SELECT * FROM $className SKIP ${skipAndLimit.skip} LIMIT ${skipAndLimit.limit}",
+              new OSQLCallback
+            )
+          )
+        }
+      } catch {
+        case ex: Exception => handleFailure(ex)
+      } finally {
+        client.close()
+      }
+    } else {
+      val oObjectClient = new OObjectDatabaseTx(client)
+      oObjectClient.getEntityManager.registerEntityClass(
+        clazz.getOrElse(throw new RuntimeException("Typed stream class is invalid"))
+      )
+
+      try {
+        if (query != null) {
+          oObjectClient.query[java.util.List[T]](new OSQLNonBlockingQuery[T](query, new OSQLCallback))
+        } else {
+          val oDocs =
+            oObjectClient
+              .query[java.util.List[T]](
+                new OSQLSynchQuery[T](
+                  s"SELECT * FROM $className SKIP ${skipAndLimit.skip} LIMIT ${skipAndLimit.limit}"
+                )
+              )
+              .asScala
+              .toList
+          if (oDocs.nonEmpty) {
+            skipAndLimit.skip = skipAndLimit.skip + skipAndLimit.limit
+          }
+          onSuccess(oDocs)
+        }
+      } catch {
+        case ex: Exception => handleFailure(ex)
+      } finally {
+        oObjectClient.close()
+        client.close()
+      }
+    }
+  }
+
+  def onFailure(exception: Exception) = failureHandler.invoke(exception)
+  def onSuccess(response: List[T]) = responseHandler.invoke(response)
+
+  def handleFailure(ex: Throwable): Unit =
+    failStage(ex)
+
+  def handleResponse(res: List[T]): Unit =
+    reader.convert(res) match {
+      case OSQLResponse(Some(error), _) =>
+        failStage(new IllegalStateException(error))
+      case OSQLResponse(None, Some(result)) if result.records.isEmpty =>
+        completeStage()
+      case OSQLResponse(_, Some(result)) =>
+        emitMultiple(out, result.records.toIterator)
+    }
+
+  setHandler(out, this)
+
+  override def onPull(): Unit = sendOSQLRequest()
+
+  class OSQLCallback extends OCommandResultListener {
+    var oDocs: List[T] = List()
+
+    override def result(iRecord: scala.Any): Boolean = {
+      oDocs :+= iRecord.asInstanceOf[T]
+      true
+    }
+
+    override def getResult: AnyRef = oDocs
+
+    override def end(): Unit = {
+      if (oDocs.nonEmpty) {
+        skipAndLimit.skip = skipAndLimit.skip + skipAndLimit.limit
+      }
+      onSuccess(oDocs)
+    }
+  }
+}
+
+private[orientdb] object SkipAndLimit {
```

Comment:
And object is static. This is dangerous because if you configure several connectors in the same application you might be innadvertedly overriding the config for one connector when creating the next.

I'd use a case class instead:
```
private[orientdb] case class SkipAndLimit (skip: Int, limit: Int)
```

if you can it to be use as `SkipAndLimit` without arguments you can add defaults
```
private[orientdb] case class SkipAndLimit (skip: Int = 0, limit: Int = 0)
```